src/main/java/busynessmanager/parser/CommandParser.java

```
@@ -0,0 +1,219 @@
+package busynessmanager.parser;
+
+public class CommandParser {
+
+    /**
+     * Constructs the CommandParser class.
+     * NOTE: temporary constructor for JUnit testing.
+     */
+    public CommandParser() {}
+
+    /**
+     * Parses the user input for execution.
+     *
+     * @param input Input from the user.
+     */
+    public void parseCommand(String input) {
+        // These two Strings are placeholders. (to be replaced by method calling in another PR)
+        String command = input.substring(0, 5);
+        String info = input.substring(5);
+
+        executeCommand(command, info);
+    }
+
+    /**
+     * Executes the appropriate method based on the command keyword.
+     *
+     * @param command Command keyword from the user input.
+     * @param info Information related to the command keyword.
+     */
+    protected void executeCommand(String command, String info) {
+        switch (command) {
+        case "add":
+            addProduct(info);
+            break;
+
+        case "delete":
+            deleteProduct(info);
+            break;
+
+        case "update":
+            updateProduct(info);
+            break;
+        case "print":
+            printProducts();
+            break;
+        case "sold":
+            recordSale(info);
+            break;
+        case "clear":
+            clearSales(info);
+            break;
+        case "revenue":
+            computeRevenue(info);
+            break;
+        case "search":
+            searchForProduct(info);
+            break;
+        default:
+            break;
```

Switch case structure is clear, but adding a default case with logging or an exception would help catch invalid commands instead of silently ignoring them.